rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java

```diff
@@ -588,83 +602,144 @@ private Integer addFederatorPublicKeyMultikey(boolean dryRun, BtcECKey btcKey, E
-            return FederationChangeResponseCode.PENDING_FEDERATION_NON_EXISTENT.getCode();
+            logger.warn("[commitFederationAccordingToActivations] Pending federation does not exist.");
+            return FederationChangeResponseCode.PENDING_FEDERATION_NON_EXISTENT;
         }
 
         if (!currentPendingFederation.isComplete()) {
-            logger.warn("[commitFederation] Pending federation has {} members, so it does not meet the minimum required.", currentPendingFederation.getMembers().size());
-            return FederationChangeResponseCode.INSUFFICIENT_MEMBERS.getCode();
+            logger.warn("[commitFederationAccordingToActivations] Pending federation has {} members, so it does not meet the minimum required.", currentPendingFederation.getMembers().size());
+            return FederationChangeResponseCode.INSUFFICIENT_MEMBERS;
         }
 
-        if (!hash.equals(currentPendingFederation.getHash())) {
-            logger.warn("[commitFederation] Provided hash {} does not match pending federation hash {}.", hash, currentPendingFederation.getHash());
-            return FederationChangeResponseCode.PENDING_FEDERATION_MISMATCHED_HASH.getCode();
+        if (!pendingFederationHash.equals(currentPendingFederation.getHash())) {
+            logger.warn("[commitFederationAccordingToActivations] Provided hash {} does not match pending federation hash {}.", pendingFederationHash, currentPendingFederation.getHash());
+            return FederationChangeResponseCode.PENDING_FEDERATION_MISMATCHED_HASH;
         }
 
         if (dryRun) {
-            logger.info("[commitFederation] DryRun execution successful.");
-            return FederationChangeResponseCode.SUCCESSFUL.getCode();
+            logger.info("[commitFederationAccordingToActivations] DryRun execution successful.");
+            return FederationChangeResponseCode.SUCCESSFUL;
         }
 
-        // Move UTXOs from the new federation into the old federation
-        // and clear the new federation's UTXOs
+        // proceed with the federation commitment
+        if (!activations.isActive(ConsensusRule.RSKIP419)) {
+            return legacyCommitFederation(currentPendingFederation, eventLogger);
+        }
+        return commitFederation(currentPendingFederation, eventLogger);
+    }
+
+    /**
+     * UTXOs are moved from active to retiring federation so that the transfer of funds can begin.
+     * Then, the retiring federation (old federation) is set to be the currently active federation,
+     * the active federation (new federation) is replaced with a new federation generated from the pending federation,
+     * and the pending federation is wiped out.
+     * The federation change info is preserved, and the commitment with the voted federation is logged.
+     */
+    private FederationChangeResponseCode legacyCommitFederation(PendingFederation currentPendingFederation, BridgeEventLogger eventLogger) {
+        moveUTXOsFromNewToOldFederation();
```

Comment:
Should this happen before setting the active fed as the old fed, and the same for the new fed?

Reply:
thats how it was before, see [previous code](https://github.com/rsksmart/rskj/blob/feature/powpeg_validation_protocol/rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java#L625-#L631)